src/navigation/actions.js

```diff
@@ -48,18 +49,25 @@ export const gotoStocktakeManagePage = ({ stocktake, stocktakeName }) =>
  *
  * @param {Object} stocktake  The requisition to navigate to.
  */
-export const gotoStocktakeEditPage = stocktake => {
+export const gotoStocktakeEditPage = stocktake => (dispatch, getState) => {
+  const { nav } = getState();
+  const currentRouteName = getCurrentRouteName(nav);
+
   const hasNegativeAdjustmentReasons = UIDatabase.objects('NegativeAdjustmentReason').length > 0;
   const hasPositiveAdjustmentReasons = UIDatabase.objects('PositiveAdjustmentReason').length > 0;
   const usesReasons = hasNegativeAdjustmentReasons && hasPositiveAdjustmentReasons;
 
-  return NavigationActions.navigate({
+  // If navigating from the stocktakesPage, go straight to the StocktakeEditPage. Otherwise,
+  // replace the current page as the user is coming from StocktakeManagePage.
+  const navigationActionCreator =
+    currentRouteName === 'stocktakes' ? NavigationActions.navigate : StackActions.replace;
```

Comment:
Just to make sure I follow, we push to the stack if navigating from `StocktakesPage`, but replace if `StocktakeManagePage`? Assuming this is so that the previous page on the stack when viewing `StocktakeEditPage` is always `StocktakesPage`?

If this is correct, suggest the comment only needs to explain this (and the rest can be seen in the code), e.g. something simple like

```
// Maintain stockTakes page at the top of the navigation stack.
// If not navigating from stocktakesPage, assume navigating from StocktakeManagePage. 
```

If incorrect, then ignore 😆 .